[ESIMD][DOC][NFC] Move ESIMD doc files from "experimental" folder to "supported" (#9892)
intel/llvm: sycl/doc/extensions/supported/sycl_ext_intel_esimd/sycl_ext_intel_esimd.md

  general register file. This allows to write efficient code not relying on
  further widening by the compiler, as with traditional SPMD programming.
- Low-level APIs efficiently mapped to the Intel GPU architecture, such as block loads/stores/gathers/scatters, explicit cache hints, GPU inline assembly, etc.
- Regular SYCL and ESIMD kernels can co-exist in the same translation unit and in

> the same application. See more details in ./README.md#

Reviewer: ```suggestion
the same application. For more details, see ./README.md#
```

Author: Nick, Thank you.
I've done fixes for this and other comments.
This line definitely deserved a fix because it had a broken link, which is fixed now.

BTW, SYCL-ESIMD-Interoperability section probably should be moved from README to the main ESIMD SPEC. I can do that a bit later.

Reviewer: cool thx

> Explicit SIMD though have some [restrictions](#implementation-restrictions) as well.

Reviewer: ```suggestion
Explicit SIMD has some [restrictions](#implementation-restrictions) as well.
```

```
more examples can be found in the
[ESIMD test suite](https://github.com/intel/llvm/tree/sycl/sycl/test-e2e/ESIMD) on github.

## Implementation restrictions

> This section contains lists of the main restrictions that apply when using the ESIMD

Reviewer: ```suggestion
This section contains a list of the main restrictions that apply when using the ESIMD
```

extension.
> **Note**: Some restrictions are not enforced by the compiler, which may lead to
> undefined program behavior if violated.

### Features not supported with the ESIMD extension:
- The [C and C++ Standard libraries support](../supported/C-CXX-StandardLibrary.rst)
- The [Device library extensions](../../../design/DeviceLibExtensions.rst)

### Unsupported standard SYCL APIs:

> Current ESIMD implementation does not support using certain standard SYCL features

Reviewer: ```suggestion
The current ESIMD implementation does not support using certain standard SYCL features
```

> inside explicit SIMD kernels and functions. Most of them will be eventually

Reviewer: ```suggestion
inside ESIMD kernels and functions. Most of them will be eventually
```
It might be a bit confusing to mix ESIMD and explicit SIMD terminology in the same sentence

> dropped. What's not supported today:

Reviewer 2: `Most of them will be eventually dropped`. What does this refer to? SYCL features below will be dropped?

Author: That sentence was wrong. Probably, the context before it was changed and the sentence remained as it was.
I fixed it in d6b2ac049189639af1f1a13eceaeb2a38b1235e4

- Interoperability between regular SYCL and ESIMD kernels is only supported one way.
  Regular SYCL kernels can call ESIMD functions, but not vice-versa. Invocation of SYCL code from ESIMD is not supported yet.
- 2D and 3D target::device accessor and local_accessor;
- Constant accessors;
- `sycl::accessor::get_pointer()` and `sycl::accessor::operator[]` are supported only with `-fsycl-esimd-force-stateless-mem`. Otherwise, All memory accesses through an accessor are
done via explicit APIs; e.g. `sycl::ext::intel::esimd::block_store(acc, offset)`
- Accessors with non-zero offsets to accessed buffer;
- Accessors with access/memory range specified;
- `sycl::image`, `sycl::sampler` and `sycl::stream` classes.

### Other restrictions:

> - Only Intel GPU device is supported.

Reviewer: ```suggestion
- Only Intel GPU devices are supported.
```

> - Interoperability between regular SYCL and ESIMD kernels is only supported one way.

Reviewer 2: This bullet has already been mentioned above.

Author: Fixed in d6b2ac049189639af1f1a13eceaeb2a38b1235e4
